### PR TITLE
React: Hide trash button next to logged device

### PR DESCRIPTION
**Pull request by cehali**
# Summary
Hide trash button nex to logged device

Fixes: 0

## Checklist
- [x] Change is small and easy to review (please split big changes into multiple PRs)
- [x] Change is consistent with architecture
- [x] Change is consistent with test architecture
- [x] Change is consistent with naming conventions
- [x] New code is covered with tests
- [x] Tests related to old code are updated
- [x] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx [hunk 1]
```diff
@@ -31,6 +32,24 @@ export const ConnectedDevicesItem = ({devicesAmount, deviceInfo, publicKey, depl
     }
   };
 
+  const isCurrentDevice = () => utils.computeAddress(deployedWallet.privateKey) === publicKey;
+
+  const renderTrashButton = () => (
+    !isCurrentDevice && (
```
Comment (rzadp) [inline universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]:
```suggestion
    !isCurrentDevice() && (
```

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx [hunk 2]
```diff
@@ -31,6 +32,24 @@ export const ConnectedDevicesItem = ({devicesAmount, deviceInfo, publicKey, depl
     }
   };
 
+  const isCurrentDevice = () => utils.computeAddress(deployedWallet.privateKey) === publicKey;
```
Comment (Marik-D) [inline universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: Extract `utils.computeAddress(deployedWallet.privateKey)` to be a prop of `DeployedWallet`

Comment (cehali) [reply universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: 👍

File: universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx [hunk 3]
```diff
@@ -31,6 +31,24 @@ export const ConnectedDevicesItem = ({devicesAmount, deviceInfo, publicKey, depl
     }
   };
 
+  const isCurrentDevice = () => deployedWallet.publicKey === publicKey;
```
Comment (Marik-D) [inline universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: this function is used only inside `renderTrashButton`, move it inside of it and maybe turn into a variable

Comment (rzadp) [reply universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: Do it in a follow-up PR? This PR is getting old

Comment (cehali) [reply universal-login-react/src/ui/UFlow/Devices/ConnectedDevicesItem.tsx]: Done

File: universal-login-sdk/lib/api/DeployedWallet.ts [hunk 4]
```diff
@@ -25,6 +25,10 @@ export class DeployedWallet implements ApplicationWallet {
   ) {
   }
 
+  get publicKey() {
+    return utils.computeAddress(this.privateKey);
+  }
+
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/lib/api/DeployedWallet.ts]: Make it one of DeployedWallet properties and assign it in the constructor. We don't need getter to one variable.

Comment (Marik-D) [reply universal-login-sdk/lib/api/DeployedWallet.ts]: What's wrong with a getter here? It seems more cleaner to have a getter for a value that's not standalone but rather derived from other fields.

Or, as an alternative, convert it to a method `getPublicKey()`

Comment (cehali) [reply universal-login-sdk/lib/api/DeployedWallet.ts]: I did it in the constructor first but after the discussion, I agree with Dima that it is better to have it as a getter.

**Pull request merged**